enterprise/backend/src/metabase_enterprise/metabot_v3/dummy_tools.clj

```diff
@@ -237,14 +237,56 @@
           []
           (:user-is-viewing context)))
 
+(defn- execute-query
+  [query-id legacy-query]
+  (let [field-id-prefix (str "field_[" query-id "]_")
+        mp (lib.metadata.jvm/application-database-metadata-provider (:database legacy-query))
+        query (lib/query mp legacy-query)]
+    {:type :query
+     :query-id query-id
+     :query legacy-query
+     :result-columns (mapv #(metabot-v3.tools.u/->result-column % field-id-prefix) (lib/returned-columns query))}))
```

Comment:
We are not really executing the query, of course. I wonder if some permissions should be checked. Read right to the database and the source table/card come to mind as possibilities.